### PR TITLE
Product Backlog Item 2124976: SaaS App Tutorial: Graph APIs to configure SAML-based single sign-on Update

### PR DESCRIPTION
**Pull request by v-hgampala**

### CODE REVIEW

**Comment by v-hgampala** [comment]
> @v-hgampala This heading " Create a signing certificate (AzureAD)" should be replaced with " Create a signing certificate generated by AzureAD". Otherwise everything else looks good to me.

@jeevansd, the heading is changed as suggested, please review it again. Thanks

**Comment by FaithOmbongi** [comment]
@AllisonAm for technical review

**Comment by jeevansd** [comment]
@FaithOmbongi I have reviewed this and looks good. @AllisonAm Can you please take a look at it?

**Comment by jeevansd** [comment]
@AllisonAm Also need your sign off here. Providing visibility to @eringreenlee as you are our doc champ here!! Yes, and you got a new badge today, Erin!!

**Comment by jeevansd** [comment]
#sign-off

**Comment by FaithOmbongi** [comment]
Hi @AllisonAm, @eringreenlee - could you complete the technical review and approve the PR so I can do the content review? Thanks.

**Comment by jeevansd** [comment]
@AllisonAm Can you please approve this PR?

File: concepts/application-saml-sso-configure-api.md [hunk 1]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
 }
 ```
 
+### Create a custom signing certificate
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:555]:
```suggestion
### Option 2: Create a custom signing certificate
```

File: concepts/application-saml-sso-configure-api.md [hunk 2]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+        {
+            var message = Encoding.ASCII.GetBytes(thumbprint);
+            SHA256Managed hashString = new SHA256Managed();
+            return Convert.ToBase64String(hashString.ComputeHash(message));
+        }
+    }
+}
+```
+
+### Add a custom signing key
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:677]: You've confirmed internally that this is required step when you use the C# code
```suggestion
#### Add a custom signing key
```

File: concepts/application-saml-sso-configure-api.md [hunk 3]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+
+Add the following information to the service principal:
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:679]:
```suggestion
After running the previous code, extract the values for the following parameters from the PFX file. You will add this information to the service principal:
```

File: concepts/application-saml-sso-configure-api.md [hunk 4]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+
+* Private key
+* Password
+* Public key 
+
+Extract the private and public key Base64 encoded from the PFX file. To learn more about the properties, see [keyCredential resource type](/graph/api/resources/keycredential?view=graph-rest-1.0&preserve-view=true).
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:685]:
```suggestion
For more information about the properties, see [keyCredential resource type](/graph/api/resources/keycredential).
```

File: concepts/application-saml-sso-configure-api.md [hunk 5]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+<!-- {
+  "blockType": "request",
+  "name": "servicePrincipals"
+}-->
+```msgraph-interactive
+PATCH https://graph.microsoft.com/v1.0/servicePrincipals/f47a6776-bca7-4f2e-bc6c-eec59d058e3e
+
+Content-type: application/json
+
+{
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:703]:
```suggestion
<!-- {
  "blockType": "request",
  "name": "tutorial_configure_saml_sso_add_credentials"
}-->
```http
PATCH https://graph.microsoft.com/v1.0/servicePrincipals/f47a6776-bca7-4f2e-bc6c-eec59d058e3e
Content-type: application/json

{
```

File: concepts/application-saml-sso-configure-api.md [hunk 6]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+
+* Private key
+* Password
+* Public key 
+
+Extract the private and public key Base64 encoded from the PFX file. To learn more about the properties, see [keyCredential resource type](/graph/api/resources/keycredential?view=graph-rest-1.0&preserve-view=true).
+
+Make sure that the keyId for the keyCredential used for "Sign" matches the keyId of the passwordCredential. You can generate the `customkeyIdentifier` by getting the hash of the cert's thumbprint. See the previous C# reference code.
+
+#### Request
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:689]:
```suggestion
##### Request
```

File: concepts/application-saml-sso-configure-api.md [hunk 7]
```diff
@@ -552,6 +552,200 @@ Content-type: application/json
...
+    ]
+}
+```
+
+#### Response
+
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+} -->
```
Comment (FaithOmbongi) [inline concepts/application-saml-sso-configure-api.md:744]:
```suggestion
##### Response

<!-- {
  "blockType": "response",
  "truncated": true
} -->
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @v-hgampala for your work on this PR. I've requested a few changes prior to approval.

**Comment by jeevansd** [comment]
Changes looks correct to me. But Allison needs to sign off here.

**Review by AllisonAm — APPROVED** [review]
Changes look good to me!

**Comment by FaithOmbongi** [comment]
Hi @v-hgampala - Allison has given her technical approval. Please address the content feedback so we can merge the doc.

**Comment by v-hgampala** [comment]
> Hi @v-hgampala - Allison has given her technical approval. Please address the content feedback so we can merge the doc.

Please merge the PR, content modified as suggested.

**Review by FaithOmbongi — APPROVED** [review]

**Pull request merged**